Orckestra.Widget.FilteredSelector/Orckestra.Widget.FilteredSelector/FunctionProvider/FilteredSelectorFunctionProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+
+using Composite.Functions;
+using Composite.Functions.Plugins.FunctionProvider;
+
+using Microsoft.Practices.EnterpriseLibrary.Common.Configuration;
+using Microsoft.Practices.EnterpriseLibrary.Common.Configuration.ObjectBuilder;
+using Microsoft.Practices.ObjectBuilder;
+
+using Orckestra.Widget.FilteredSelector.FunctionProvider.Foundation;
+using Orckestra.Widget.FilteredSelector.FunctionProvider.Functions;
+
+namespace Orckestra.Widget.FilteredSelector.FunctionProvider
+{
+    [ConfigurationElementType(typeof(FilteredSelectorFunctionProviderData))]
+    public class FilteredSelectorFunctionProvider : IFunctionProvider
+    {
+        private readonly EntityTokenFactory _entityTokenFactory;
+        private List<IFunction> _functions = null;
+        public FunctionNotifier FunctionNotifier { set { } }
+
+        public IEnumerable<IFunction> Functions
+        {
+            get
+            {
+                if (_functions == null)
+                {
+                    InitializeStaticTypeFunctions();
+                }
+
+                foreach (IFunction function in _functions)
+                {
+                    yield return function;
+                }

```

Comment:
"yield return" and for are redundant, you can just use

`return _functions;`

Reply:
Updated